enterprise/tailnet/pgcoord.go

```diff
@@ -106,30 +107,74 @@ func NewPGCoord(ctx context.Context, logger slog.Logger, ps pubsub.Pubsub, store
 	id := uuid.New()
 	logger = logger.Named("pgcoord").With(slog.F("coordinator_id", id))
 	bCh := make(chan binding)
-	cCh := make(chan *connIO)
+	cCh := make(chan agpl.Queue)
+	sCh := make(chan subscribe)
 	// signals when first heartbeat has been sent, so it's safe to start binding.
 	fHB := make(chan struct{})
 
 	c := &pgCoord{
-		ctx:            ctx,
-		cancel:         cancel,
-		logger:         logger,
-		pubsub:         ps,
-		store:          store,
-		binder:         newBinder(ctx, logger, id, store, bCh, fHB),
-		bindings:       bCh,
-		newConnections: cCh,
-		id:             id,
-		querier:        newQuerier(ctx, logger, ps, store, id, cCh, numQuerierWorkers, fHB),
-		closed:         make(chan struct{}),
+		ctx:              ctx,
+		cancel:           cancel,
+		logger:           logger,
+		pubsub:           ps,
+		store:            store,
+		binder:           newBinder(ctx, logger, id, store, bCh, fHB),
+		bindings:         bCh,
+		newConnections:   cCh,
+		newSubscriptions: sCh,
+		id:               id,
+		querier:          newQuerier(ctx, logger, id, ps, store, id, cCh, sCh, numQuerierWorkers, fHB),
+		closed:           make(chan struct{}),
 	}
 	logger.Info(ctx, "starting coordinator")
 	return c, nil
 }
 
-func (c *pgCoord) ServeMultiAgent(id uuid.UUID) agpl.MultiAgentConn {
-	_, _ = c, id
-	panic("not implemented") // TODO: Implement
+func (c *pgCoord) ServeMultiAgent(id uuid.UUID) (agpl.MultiAgentConn, error) {
+	ma := (&agpl.MultiAgent{
+		ID:                id,
+		AgentIsLegacyFunc: func(agentID uuid.UUID) bool { return true },
+		OnSubscribe: func(enq agpl.Queue, agent uuid.UUID) (*agpl.Node, error) {
+			err := sendCtx(c.ctx, c.newSubscriptions, subscribe{
+				q:       enq,
+				agentID: agent,
+				active:  true,
+			})
+
+			return c.Node(agent), err
+		},
+		OnUnsubscribe: func(enq agpl.Queue, agent uuid.UUID) error {
+			return sendCtx(c.ctx, c.newSubscriptions, subscribe{
+				q:       enq,
+				agentID: agent,
+				active:  false,
+			})
+		},
+		OnNodeUpdate: func(id uuid.UUID, node *agpl.Node) error {
+			return sendCtx(c.ctx, c.bindings, binding{
+				bKey: bKey{id, agpl.QueueKindClient},
+				node: node,
+			})
+		},
+		OnRemove: func(enq agpl.Queue) {
+			b := binding{
+				bKey: bKey{
+					id:   enq.UniqueID(),
+					kind: enq.Kind(),
+				},
+			}
+			if err := sendCtx(c.ctx, c.bindings, b); err != nil {
+				c.logger.Debug(c.ctx, "parent context expired while withdrawing bindings", slog.Error(err))
+			}
+			c.querier.cleanupConn(enq)
+		},
+	}).Init()
+
+	if err := sendCtx(c.ctx, c.newConnections, agpl.Queue(ma)); err != nil {
+		return nil, err
```

Comment:
I think you can handle this error gracefully instead of returning it.  That will save a lot of code changes across files in this PR.

`sendCtx` returns an error when the context is expired/canceled, which means the PGCoord is shutting down.  You can just return a closed MultiAgent in this case.  Probably we're shutting down all of Coderd, so it won't get used anyway, but if it does, MultiAgent returns "closed" errors just like if the close happened after `ServeMultiAgent()` returns.